modules/web/client/src/main/scala/observe/ui/components/sequence/SequenceTable.scala

```diff
@@ -284,16 +287,51 @@ private sealed trait SequenceTableBuilder[S: Eq, D <: DynamicConfig: Eq]
                 case _                                    =>
                   TagMod.empty
 
-        val (icon, label) =
-          if table.getIsAllRowsExpanded() then (Icons.Minus, "Collapse all visits")
-          else (Icons.Plus, "Expand all visits")
+        def modExpanded(
+          f: (
+            Row[Expandable[HeaderOrRow[SequenceIndexedRow[D]]]]
+          ) => Boolean
+        ): Callback = Callback.suspend:
+          table.setExpanded:
+            Expanded.Rows(
+              table
+                .getExpandedRowModel()
+                .rows
+                .map: row =>
+                  RowId(row.id) -> f(Row(row))
+                .toMap
+            )
+
+        val allVisitsAreExpanded = table
```

Comment:
Similarly, repetition could be avoided by extracting a method that checks if all visits have expanded == (true | false).

Reply:
I've refactored it a little bit to reduce code reuse